### DISCUSSION TITLE
Picking a preferred dependency manager for our documentation.

### DISCUSSION THREAD

**Discussion by Richard-Gist** [wwt/SwiftCurrent#40] [Ideas] [answered]
As brought up in this PR: https://github.com/wwt/Workflow/pull/37#discussion_r649593619

> We should have a conversation about what our preferred package manager is. Most of our instructions reference CocoaPods but it seems maybe more likely our preferred package manager should be SwiftPM.

Right now we are writing as if CocoaPods is our preferred dependency manager, but we also support SPM.  Let's discuss here on the merits of both and which one we should primarily write for.

**Comment by Richard-Gist** [comment]
Initially I was for CocoaPods, because from my bubble it seemed that CocoaPods was the popular and preferred DM.  After some talks and reflecting, I'm now on SPM.

My reasonings are:

1. SPM is the future and direction of Apple and will continue to be developed by them.
2. As it grows, it will become the standard within iOS and Swift.
3. And my understanding is that if you were using Carthage, you would now use SPM, potentially making that group larger than CocoaPods.

I thought these were compelling enough to switch from my flawed assumption that CocoaPods was the most popular dependency manager.

+1 to SPM

**Comment by Tyler-Keith-Thompson** [comment]
I also believe SPM should be our preferred package manager. I agree with all of your points and have some additional ones:

1. Preferring SPM in our docs aligns us more with how Apple is positioned with all of their open source projects. Generally I feel they're good to emulate, with some exceptions.
2. It's actually grown quite a bit already, It's over 5 years old already.
3. In typical Apple form, it's becoming so heavily integrated with other tooling that it's almost unthinkable not to use it already.
4. It's supported by paid Apple Employees AND the community, meaning work is constantly happening with it.

A plug for CocoaPods:
CocoaPods is a more mature package manager, it has more fully fledged features, like subspecs, tests included with your lib, and example apps you can include with your lib. That being said SPM is catching up fast, and CocoaPods themselves have a section on their blog saying they're actively watching SPM and will consider stopping development as it becomes as feature rich as they are.

**Comment by morganzellers** [comment]
I think Tyler's reason 3 encapsulates my main reason, but I agree with all other points made above.

+1 SPM

**Comment by morganzellers** [comment, marked as answer]
Decided on SPM as our preferred package manager.
Summarized reasoning:

1. SPM is the future and direction of Apple and will continue to be developed by them.
2. As it grows, it will become the standard within iOS and Swift.
3. And my understanding is that if you were using Carthage, you would now use SPM, potentially making that group larger than CocoaPods.
4. Preferring SPM in our docs aligns us more with how Apple is positioned with all of their open source projects. Generally I feel they're good to emulate, with some exceptions.
5. It's actually grown quite a bit already, It's over 5 years old already.
6. In typical Apple form, it's becoming so heavily integrated with other tooling that it's almost unthinkable not to use it already.
7. It's supported by paid Apple Employees AND the community, meaning work is constantly happening with it.
